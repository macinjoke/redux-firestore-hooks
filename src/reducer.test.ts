import { createReducer, clear, collectionUpdated, documentUpdated } from '.'

test('collectionUpdated', () => {
  const payload1 = {
    key: 'foo',
    dataGroup: {
      x: { x1: 'x1 text', x2: 11 },
      y: { y1: true, y2: null },
    },
  }
  const payload2 = {
    key: 'foo',
    dataGroup: {},
  }
  const reducer = createReducer()

  const initialState = {}

  const state1 = reducer(initialState, collectionUpdated(payload1))
  expect(state1).toMatchSnapshot()
  const state2 = reducer(state1, collectionUpdated(payload2))
  expect(state2).toMatchSnapshot()
})

test('documentUpdated', () => {
  const payload1 = {
    key: 'foo',
    id: 'id1',
    data: {
      x1: 'x1 text',
      x2: 11,
    },
  }
  const payload2 = {
    key: 'foo',
    id: 'id1',
    data: {},
  }
  const reducer = createReducer()

  const initialState = {}

  const state1 = reducer(initialState, documentUpdated(payload1))
  expect(state1).toMatchSnapshot()
  const state2 = reducer(state1, documentUpdated(payload2))
  expect(state2).toMatchSnapshot()
})

test('clear', () => {
  const reducer = createReducer()

  const initialState = {
    foo: {
      x: {
        x1: 'x1 text',
        x2: 11,
      },
      y: {
        y1: true,
      },
      z: {
        z1: null,
      },
    },
  }

  const state1 = reducer(initialState, clear(['foo', 'x']))
  expect(state1).toMatchSnapshot()
  const state2 = reducer(state1, clear('foo'))
  expect(state2).toMatchSnapshot()
})
