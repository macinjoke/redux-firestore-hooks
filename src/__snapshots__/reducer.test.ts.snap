// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clear 1`] = `
Object {
  "foo": Object {
    "y": Object {
      "y1": true,
    },
    "z": Object {
      "z1": null,
    },
  },
}
`;

exports[`clear 2`] = `Object {}`;

exports[`collectionUpdated 1`] = `
Object {
  "foo": Object {
    "x": Object {
      "x1": "x1 text",
      "x2": 11,
    },
    "y": Object {
      "y1": true,
      "y2": null,
    },
  },
}
`;

exports[`collectionUpdated 2`] = `
Object {
  "foo": Object {},
}
`;

exports[`documentUpdated 1`] = `
Object {
  "foo": Object {
    "id1": Object {
      "x1": "x1 text",
      "x2": 11,
    },
  },
}
`;

exports[`documentUpdated 2`] = `
Object {
  "foo": Object {
    "id1": Object {},
  },
}
`;
